Quarkus 3.26.1 Native Build fails

Java version: 21.0.8+9-LTS, vendor version: Mandrel-23.1.8.0-Final
 Graal compiler: optimization level: 2, target machine: x86-64-v3
 C compiler: gcc (redhat, x86_64, 11.5.0)


```
Error: Discovered unresolved method during parsing: org.apache.commons.compress.compressors.zstandard.ZstdCompressorInputStream.<init>(java.io.InputStream). This error is reported at image build time because class org.apache.commons.compress.archivers.zip.ZipArchiveInputStream is registered for linking at image build time by command line and command line.
Error encountered while parsing org.apache.commons.compress.archivers.zip.ZipArchiveInputStream.createZstdInputStream(ZipArchiveInputStream.java:525) 
Parsing context:
   at org.apache.commons.compress.archivers.zip.ZipArchiveInputStream.getNextZipEntry(ZipArchiveInputStream.java:794)
   at org.apache.tika.detect.zip.DefaultZipContainerDetector.detectStreaming(DefaultZipContainerDetector.java:297)
   at org.apache.tika.detect.zip.DefaultZipContainerDetector.tryStreamingOnTikaInputStream(DefaultZipContainerDetector.java:210)
   at org.apache.tika.detect.zip.DefaultZipContainerDetector.detect(DefaultZipContainerDetector.java:191)
   at org.apache.tika.detect.CompositeDetector.detect(CompositeDetector.java:84)
   at org.apache.tika.Tika.detect(Tika.java:160)
   at org.apache.tika.Tika.detect(Tika.java:289)
```

Unfortunately this did not work. Build works now but on runtime I get the error.

```
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-compress</artifactId>
            <version>1.27.1</version>
        </dependency>
        <dependency>
            <groupId>io.quarkiverse.tika</groupId>
            <artifactId>quarkus-tika</artifactId>
            <version>2.2.0</version>
            <exclusions>
                <exclusion>
                    <groupId>org.apache.commons</groupId>
                    <artifactId>commons-compress</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```


```
java.lang.RuntimeException: org.apache.commons.compress.archivers.ArchiveException: No Archiver found for the stream signature
	at io.quarkus.vertx.runtime.VertxEventBusConsumerRecorder.wrapIfNecessary(VertxEventBusConsumerRecorder.java:211)
	at io.quarkus.vertx.runtime.VertxEventBusConsumerRecorder$3$1$2.call(VertxEventBusConsumerRecorder.java:154)
	at io.quarkus.vertx.runtime.VertxEventBusConsumerRecorder$3$1$2.call(VertxEventBusConsumerRecorder.java:146)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$4(ContextImpl.java:192)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
	at io.vertx.core.impl.ContextImpl$1.execute(ContextImpl.java:221)
	at io.vertx.core.impl.WorkerTask.run(WorkerTask.java:56)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:645)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@21.0.8/java.lang.Thread.runWith(Thread.java:1596)
	at java.base@21.0.8/java.lang.Thread.run(Thread.java:1583)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
Caused by: org.apache.commons.compress.archivers.ArchiveException: No Archiver found for the stream signature
	at org.apache.commons.compress.archivers.ArchiveStreamFactory.detect(ArchiveStreamFactory.java:295)
	at org.apache.tika.detect.zip.DefaultZipContainerDetector.detectArchiveFormat(DefaultZipContainerDetector.java:122)
	at org.apache.tika.detect.zip.DefaultZipContainerDetector.detect(DefaultZipContainerDetector.java:180)
	at org.apache.tika.detect.CompositeDetector.detect(CompositeDetector.java:84)
	at org.apache.tika.Tika.detect(Tika.java:158)
	at org.apache.tika.Tika.detect(Tika.java:289)
```

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -95,18 +95,6 @@
                     </plugin>
                 </plugins>
             </build>
-            <dependencies>
-                <dependency>
-                    <groupId>com.github.luben</groupId>
-                    <artifactId>zstd-jni</artifactId>
-                    <version>1.5.7-4</version>
-                </dependency>
-                <dependency>
-                    <groupId>org.tukaani</groupId>
-                    <artifactId>xz</artifactId>
-                    <version>1.10</version>
-                </dependency>
-            </dependencies>
             <properties>
                 <quarkus.package.type>native</quarkus.package.type>
                 <quarkus.native.additional-build-args>

--- a/integration-tests/src/main/java/io/quarkus/it/tika/TikaDetectResource.java
+++ b/integration-tests/src/main/java/io/quarkus/it/tika/TikaDetectResource.java
@@ -1,0 +1,27 @@
+package io.quarkus.it.tika;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.apache.tika.Tika;
+
+@Path("/detect")
+public class TikaDetectResource {
+    // Avoiding the injection, otherwise the recorded tika-config.xml intended for TikaPdfInvoiceTest is used
+    Tika tika = new Tika();
+
+    @POST
+    @Path("/text")
+    @Consumes({ "text/plain", "application/pdf", "application/vnd.oasis.opendocument.text" })
+    @Produces(MediaType.TEXT_PLAIN)
+    public String extractText(InputStream stream) throws IOException {
+        return tika.detect(stream);
+    }
+
+}

--- a/integration-tests/src/test/java/io/quarkus/it/tika/NativeTikaDetectIT.java
+++ b/integration-tests/src/test/java/io/quarkus/it/tika/NativeTikaDetectIT.java
@@ -1,0 +1,7 @@
+package io.quarkus.it.tika;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+
+@QuarkusIntegrationTest
+public class NativeTikaDetectIT extends TikaDetectTest {
+}

--- a/integration-tests/src/test/java/io/quarkus/it/tika/TikaDetectTest.java
+++ b/integration-tests/src/test/java/io/quarkus/it/tika/TikaDetectTest.java
@@ -1,0 +1,49 @@
+package io.quarkus.it.tika;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.startsWith;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class TikaDetectTest {
+
+    @Test
+    public void testGetTextFromTextFormat() throws Exception {
+        checkText("text/plain", "txt");
+    }
+
+    @Test
+    public void testGetTextFromOdtFormat() throws Exception {
+        checkText("application/vnd.oasis.opendocument.text", "odt");
+    }
+
+    @Test
+    public void testGetTextFromPdfFormat() throws Exception {
+        checkText("application/pdf", "pdf");
+    }
+
+    @Test
+    public void testGetTextFromPdfFormatWithFonts() throws Exception {
+        given()
+                .when().header("Content-Type", "application/pdf")
+                .body(getClass().getClassLoader().getResourceAsStream("americanexpress.pdf"))
+                .post("/detect/text")
+                .then()
+                .statusCode(200)
+                .body(startsWith("application/pdf"));
+    }
+
+    private void checkText(String contentType, String extension) throws Exception {
+        given()
+                .when().header("Content-Type", contentType)
+                .body(getClass().getClassLoader().getResourceAsStream("quarkus." + extension).readAllBytes())
+                .post("/detect/text")
+                .then()
+                .statusCode(200)
+                .body(is(contentType));
+    }
+}

--- a/pom.xml
+++ b/pom.xml
@@ -146,6 +146,16 @@
                 <artifactId>commons-compress</artifactId>
                 <version>1.28.0</version>
             </dependency>
+            <dependency>
+                <groupId>com.github.luben</groupId>
+                <artifactId>zstd-jni</artifactId>
+                <version>1.5.7-4</version>
+            </dependency>
+            <dependency>
+                <groupId>org.tukaani</groupId>
+                <artifactId>xz</artifactId>
+                <version>1.10</version>
+            </dependency>
         </dependencies>
     </dependencyManagement>
     <build>

--- a/runtime/pom.xml
+++ b/runtime/pom.xml
@@ -118,6 +118,14 @@
             <artifactId>log4j2-jboss-logmanager</artifactId>
         </dependency>
         <dependency>
+            <groupId>com.github.luben</groupId>
+            <artifactId>zstd-jni</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.tukaani</groupId>
+            <artifactId>xz</artifactId>
+        </dependency>
+        <dependency>
             <groupId>org.graalvm.sdk</groupId>
             <artifactId>graal-sdk</artifactId>
             <scope>provided</scope>

